### ISSUE TITLE
Captured image dimensions shown inside the capture area

### ISSUE THREAD

**Issue by komoricodrutz** [flameshot-org/flameshot#3207]
### Feature Description

Hi,
I've been happily using flameshot for years now and noticed this small inconvenience, but was always in a hurry
This is not really bug or feature request per se, it's more like a usage issue:
![image](https://github.com/flameshot-org/flameshot/assets/134259571/9cfeed02-09e6-4ca1-a44e-80b9d380443a)

As can be seen from the screenshot above, the small area with the image dimentions and coordinates is shown inside the screen capture area, in the lower-right corner. This makes a precise screenshot for that part of the capture virtually impossible. The thing gets even worse if the captured area is very small, as the image details text ends up covering the entire capture area.
It is not impossible to see, but very imprecise and time-consuming, because if you want to be precise, you have to resize the capture area, wait for the image details to disappear, then nudge that lower-right corner a few pixels and repeat ad nauseam until you are somewhat satisfied with the result.
Would it be possible to move the image details somewhere outside the captured area? Like below the additional buttons, etc.

Thank you and keep up the brilliant work

**Comment by mmahmoudian** [comment]
> This makes a precise screenshot for that part of the capture virtually impossible.

The information should be display somewhere, right? But we have thought about that too. You have now two things at your disposal to solve this:
1. In the configurations you can make this to only appear in the middle of the region, so the corners are not affected
2. For pixel-perfect selection, you can activate magnifiers from the config

Does these address your issue?

**Comment by komoricodrutz** [comment]
> > This makes a precise screenshot for that part of the capture virtually impossible.
> 
> The information should be display somewhere, right? But we have thought about that too. You have now two things at your disposal to solve this:
> 
>     1. In the configurations you can make this to only appear in the middle of the region, so the corners are not affected
> 
>     2. For pixel-perfect selection, you can activate magnifiers from the config
> 
> 
> Does these address your issue?
Absolutely, I agree that the info should be displayed!
I believe the easiest solution is to place it just below the other round buttons.
If you place it in the middle of the capture area, it will still affect you if you are trying to capture a very small area. (it actually already does. Just try taking a screenshot of just your own avatar 
![image](https://github.com/flameshot-org/flameshot/assets/134259571/4cdca6eb-7103-4ac8-ae26-bf739b5e38f0)
here and you'll see what I mean.
Or. another workaround would to make the display of this information optional (some on/off toggle somewhere in the settings).
Thanks for the quick reply!

**Comment by komoricodrutz** [comment]
> > This makes a precise screenshot for that part of the capture virtually impossible.
> 
> The information should be display somewhere, right? But we have thought about that too. You have now two things at your disposal to solve this:
> 
>     1. In the configurations you can make this to only appear in the middle of the region, so the corners are not affected
> 
>     2. For pixel-perfect selection, you can activate magnifiers from the config
> 
> 
> Does these address your issue?
Absolutely, I agree that the info should be displayed!
I believe the easiest solution is to place it just below the other round buttons.
If you place it in the middle of the capture area, it will still affect you if you are trying to capture a very small area. (it actually already does. Just try taking a screenshot of just your own avatar 
![image](https://github.com/flameshot-org/flameshot/assets/134259571/4cdca6eb-7103-4ac8-ae26-bf739b5e38f0)
here and you'll see what I mean.
Or. another workaround would to make the display of this information optional (some on/off toggle somewhere in the settings).
Thanks for the quick reply!

**Comment by mmahmoudian** [comment]
>  it will still affect you if you are trying to capture a very small area.

True, but I would consider that as an edge case. Extremely rarely people would screenshot something that is that small, and if they want to do that, perhaps they should use the magnifier. Fir instance in the following I am dragging the bottom-right handle:

![image](https://github.com/flameshot-org/flameshot/assets/390889/1a807d60-a5d7-4484-8785-38ef72157768)


> another workaround would to make the display of this information optional (some on/off toggle somewhere in the settings).

Already possible 👇🏼 

![image](https://github.com/flameshot-org/flameshot/assets/390889/367a50a5-3222-47a9-b13a-87e6c5826021)

**Comment by komoricodrutz** [comment]
> > it will still affect you if you are trying to capture a very small area.
> 
> True, but I would consider that as an edge case. Extremely rarely people would screenshot something that is that small, and if they want to do that, perhaps they should use the magnifier. Fir instance in the following I am dragging the bottom-right handle:
> 
> ![image](https://user-images.githubusercontent.com/390889/241406290-1a807d60-a5d7-4484-8785-38ef72157768.png)
> 
> > another workaround would to make the display of this information optional (some on/off toggle somewhere in the settings).
> 
> Already possible 👇🏼
> 
> ![image](https://user-images.githubusercontent.com/390889/241406068-367a50a5-3222-47a9-b13a-87e6c5826021.png)

Damn! I was too blind to see this... Shoot me! Maybe the term "Geometry Display" was also not intuitive enough for me. (maybe you could add a tooltip on hover in the future, to actually explain what that function does?)
The magnifer option is also a bit too laborious, but the feature of simply disabling the Geometry Display is the best solution in this context.
Would be nice, however, to have the best of both worlds... So an option to display it OUTSIDE the capture area, as mentioned, somewhere below the other round buttons, would be a welcomed future feature, I think.
Thank you for making a blind man see. :-)
I think we can consider this topic closed...

**Comment by trathborne** [comment]
I'm using flameshot-12.1.0-1.ubuntu-22.04.amd64.deb on X11 (Ubuntu 22.04).

It's weird. I don't have the Configuration/General/Geometry Display section at all. Undo Limit and Imgur and then Configuration File at the bottom. I also assigned Z to "Show X and Y dimensions of the selection" and it does nothing.

What am I missing?

**Comment by mmahmoudian** [comment]
@trathborne 

> What am I missing?

Short answer: The unstable release
Long answer: This feature is not released yet. It was added via #2766 which is part of [the version13 features](https://github.com/flameshot-org/flameshot/milestone/7).

if you want to try it, you can install the latest development build:

https://flameshot.org/docs/installation/development-build

**Comment by komoricodrutz** [comment]
> I'm using flameshot-12.1.0-1.ubuntu-22.04.amd64.deb on X11 (Ubuntu 22.04).
> 
> It's weird. I don't have the Configuration/General/Geometry Display section at all. Undo Limit and Imgur and then Configuration File at the bottom. I also assigned Z to "Show X and Y dimensions of the selection" and it does nothing.
> 
> What am I missing?

Odd...
I have
Flameshot v12.1.0 ()
Compiled with Qt 5.12.8
linux: 5.15.0-76-generic
linuxmint: 20.3
And the option was there and worked.

**Comment by mmahmoudian** [comment]
That is odd, because the [v12.1.0 was released in 2022-07-03](https://github.com/flameshot-org/flameshot/releases/tag/v12.1.0) and the #2766 was merged into master branch on 2022-07-10 which is almost one week after the release of v12.1.0

**Comment by Andrew15-5** [comment]
I definitely remember that this information was displayed outside before, and I was happy, but now it really annoys me that it's inside the selection area. And I use this as a ruler a lot of times, so almost all of these times I need to select a thin area, which makes it impossible to capture the correct size and/or see the numbers. I can set `showSelectionGeometryHideTime`, which sometimes helps with actually seeing what is selected, but also makes it even harder to see what was the dimensions after the last resize.

I want an option that will put the info outside of the selection area (maybe with some padding). I don't really care on its position as long as it is outside and visible on the screen. Which actually means that in the edge cases where you capture a full screen, it has to switch back to displaying inside the area. I don't know if it's hard to implement this detection. Probably not?

**Comment by mmahmoudian** [comment]
@Andrew15-5 

> I don't know if it's hard to implement this detection. Probably not?

It is actually not easy at all. In Wayland the margin between monitors are not reported (as far as I know), and Flameshot see a canvas as large as a square that can engulf all monitors. This means that in multi-monitor setup we cannot detect if the selected region is touching the edges or is outside of the monitor.

If my understanding is right, this needs careful investigation on what type of information we can get from the OS and how can we use them.

I will keep this issue open, although I don't know how fast this would be addressed.

**Comment by BaconCatBug** [comment]
I too would like to see the geometry data be either on the outside or an option to toggle inside and outside, it makes no sense for it to be on the inside but it would be nice for people to have the option
